router/internal/retrytransport/retry_transport.go

```diff
@@ -101,6 +104,20 @@ func (rt *RetryHTTPTransport) RoundTrip(req *http.Request) (*http.Response, erro
 	return resp, err
 }
 
+func (rt *RetryHTTPTransport) drainBody(resp *http.Response) {
+	if resp == nil || resp.Body == nil {
+		return
+	}
+	_, err := io.Copy(io.Discard, resp.Body)
```

Comment:
Do we need the extra step to discard the content? Shouldn't close be sufficient?

Reply:
Yes, from my testing closing the body didn't make go mark `reusable=true`, but reading the entire body, even without closing did.